tests/Unit/DataStructures/Tensor/EagerMath/Test_Norms.cpp

```diff
@@ -0,0 +1,204 @@
+  CHECK(get(db::get<Tags::PointwiseL2Norm<Vector<3, Frame>>>(box)) ==
+        DataVector{npts, 7.0710678118654755});
+  // rank (0, 1)
+  CHECK(get(db::get<Tags::PointwiseL2Norm<Covector<1, Frame>>>(box)) ==
+        DataVector{npts, 4.});
+  CHECK(get(db::get<Tags::PointwiseL2Norm<Covector<2, Frame>>>(box)) ==
+        DataVector{npts, 4.47213595499958});
+  CHECK(get(db::get<Tags::PointwiseL2Norm<Covector<3, Frame>>>(box)) ==
+        DataVector{npts, 12.806248474865697});
+  // rank (0, 2)
+  CHECK(get(db::get<Tags::PointwiseL2Norm<Metric<1, Frame>>>(box)) ==
+        DataVector{npts, 5.});
+  CHECK(get(db::get<Tags::PointwiseL2Norm<Metric<2, Frame>>>(box)) ==
+        DataVector{npts, 3.605551275463989});
+  CHECK(get(db::get<Tags::PointwiseL2Norm<Metric<3, Frame>>>(box)) ==
+        DataVector{npts, 6.324555320336759});
+  // rank (2, 0)
+  CHECK(get(db::get<Tags::PointwiseL2Norm<InverseMetric<1, Frame>>>(box)) ==
+        DataVector{npts, 0.2});
+  CHECK(get(db::get<Tags::PointwiseL2Norm<InverseMetric<2, Frame>>>(box)) ==
+        DataVector{npts, 1.8027756377319946});
+  CHECK(get(db::get<Tags::PointwiseL2Norm<InverseMetric<3, Frame>>>(box)) ==
+        DataVector{npts, 0.4787135538781691});
+
+  // Test L2-norm reduced over domain, against precomputed values
+  // rank 0
+  CHECK(db::get<Tags::L2Norm<MyScalar>>(box) == approx(3.));
+  // rank (1, 0)
+  CHECK(db::get<Tags::L2Norm<Vector<1, Frame>>>(box) == approx(3.));
+  CHECK(db::get<Tags::L2Norm<Vector<2, Frame>>>(box) ==
+        approx(4.242640687119285));
+  CHECK(db::get<Tags::L2Norm<Vector<3, Frame>>>(box) ==
+        approx(7.0710678118654755));
+  // rank (0, 1)
+  CHECK(db::get<Tags::L2Norm<Covector<1, Frame>>>(box) == approx(4.));
+  CHECK(db::get<Tags::L2Norm<Covector<2, Frame>>>(box) ==
+        approx(4.47213595499958));
+  CHECK(db::get<Tags::L2Norm<Covector<3, Frame>>>(box) ==
+        approx(12.806248474865697));
+  // rank (0, 2)
+  CHECK(db::get<Tags::L2Norm<Metric<1, Frame>>>(box) == approx(5.));
+  CHECK(db::get<Tags::L2Norm<Metric<2, Frame>>>(box) ==
+        approx(3.605551275463989));
+  CHECK(db::get<Tags::L2Norm<Metric<3, Frame>>>(box) ==
+        approx(6.324555320336759));
+  // rank (2, 0)
+  CHECK(db::get<Tags::L2Norm<InverseMetric<1, Frame>>>(box) == approx(0.2));
+  CHECK(db::get<Tags::L2Norm<InverseMetric<2, Frame>>>(box) ==
+        approx(1.8027756377319946));
+  CHECK(db::get<Tags::L2Norm<InverseMetric<3, Frame>>>(box) ==
+        approx(0.4787135538781691));
+
+  using Tag = MyScalar;
+  /// [l2norm_name]
+  CHECK(Tags::PointwiseL2Norm<Tag>::name() ==
+        "PointwiseL2Norm(" + db::tag_name<Tag>() + ")");
+  CHECK(Tags::L2Norm<Tag>::name() == "L2Norm(" + db::tag_name<Tag>() + ")");
+  /// [l2norm_name]
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.DataStructures.Tensor.EagerMath.Norms",
+                  "[DataStructures][Unit]") {
+  test_l2_norm_tag<Frame::Grid>();
+  test_l2_norm_tag<Frame::Inertial>();
```

Comment:
Does the `Frame` actually change anything?

Reply:
This repeats the test for tensors in another frame, and the result do / should not change when compared against hard-coded numbers, which is the point. Testing for 2 frames shows that the L2-norm computation is agnostic to frames.

Reply:
Does it really test that it's agnostic to frames? For that you would have to take numbers in one frame and compute norms, then coordinate-transform those values and compute norms with the transformed values. And if I'm not mistaken that test is not expected to pass. Or perhaps I misunderstand what you mean with "agnostic to frames".

Reply:
I think this is testing that the same numerical inputs produce the same numerical outputs, regardless of the Frame parameter. In this sense (!= the physics sense), the norm is independent of Frame. Because the norm doesn't know about the metric, it should give a different answer if the tensors are transformed into a different Frame, where their components will have different numerical values.

Reply:
Right. Since nothing actually uses the frame parameter I'm not sure if this adds anything to the test.

Reply:
Exactly as @fmahebert said, this bit only tests that the same numerical input gives the same numerical output, irrespective of the frame parameter. In other words, it confirms that "nothing actually uses the frame parameter".  That is all it adds to the tests.